presto-main/src/test/java/com/facebook/presto/operator/scalar/TestDateTimeFunctions.java

```diff
@@ -148,29 +154,102 @@ private static long epochDaysInZone(TimeZoneKey timeZoneKey, long instant)
     public void testLocalTime()
             throws Exception
     {
-        long millis = new LocalTime(session.getStartTime(), DateTimeZone.UTC).getMillisOfDay();
-        functionAssertions.assertFunction("LOCALTIME", TimeType.TIME, toTime(millis));
+        Session localSession = testSessionBuilder()
+                .setSystemProperty("legacy_timestamp", "false")
+                .setTimeZoneKey(TIME_ZONE_KEY)
+                .setStartTime(new DateTime(2017, 3, 1, 10, 0, 0, 0, DateTimeZone.UTC).getMillis())
+                .build();
+        FunctionAssertions localAssertion = new FunctionAssertions(localSession);
+        localAssertion.assertFunctionString("LOCALTIME", TimeType.TIME, "15:45:00.000");
     }
 
     @Test
     public void testCurrentTime()
             throws Exception
     {
-        long millis = new LocalTime(session.getStartTime(), DateTimeZone.UTC).getMillisOfDay();
-        functionAssertions.assertFunction("CURRENT_TIME", TIME_WITH_TIME_ZONE, new SqlTimeWithTimeZone(millis, session.getTimeZoneKey()));
+        Session localSession = testSessionBuilder()
+                .setSystemProperty("legacy_timestamp", "false")
+                .setTimeZoneKey(TIME_ZONE_KEY)
+                .setStartTime(new DateTime(2017, 3, 1, 10, 0, 0, 0, DateTimeZone.UTC).getMillis())
+                .build();
+        FunctionAssertions localAssertion = new FunctionAssertions(localSession);
+        localAssertion.assertFunctionString("CURRENT_TIME", TIME_WITH_TIME_ZONE, "15:45:00.000 Asia/Kathmandu");
     }
 
     @Test
     public void testLocalTimestamp()
     {
-        functionAssertions.assertFunction("localtimestamp", TimestampType.TIMESTAMP, toTimestamp(session.getStartTime()));
+        Session localSession = testSessionBuilder()
+                .setSystemProperty("legacy_timestamp", "false")
+                .setTimeZoneKey(TIME_ZONE_KEY)
+                .setStartTime(new DateTime(2017, 3, 1, 10, 0, 0, 0, DateTimeZone.UTC).getMillis())
+                .build();
+        FunctionAssertions localAssertion = new FunctionAssertions(localSession);
```

Comment:
`FunctionsAssertions` are closeable (and contain `LocalQueryRunner`) and should be closed.